### PR TITLE
 Integrate stateless hierarchies builder into test app

### LINKED ISSUE #283: Integrate into test app
Need to integrate stateless hierarchies builder into our test app for testing it.

### PR DESCRIPTION
**Pull request by JonasDov**
closes #283

### CODE REVIEW

File: apps/test-app/frontend/src/components/tree-widget/TreeWidget.tsx [hunk 1]
```diff
@@ -52,14 +87,17 @@ export function TreeWidget(props: Props) {
     );
     setMatchesCount(newMatchesCount);
   }, []);
-
   const { width, height, ref } = useResizeDetector();
```
Comment (saskliutas) [inline apps/test-app/frontend/src/components/tree-widget/TreeWidget.tsx:90]: It should be enough to track size in `TreeWidget` component and pass it down to rules driven and stateless trees.

File: apps/test-app/frontend/src/components/tree-widget/TreeWidget.tsx [hunk 2]
```diff
@@ -32,16 +33,50 @@ import { Tree } from "./Tree";
...
   rulesetId?: string;
+  height?: number;
 }
 
 export function TreeWidget(props: Props) {
+  const [openTab, setOpenTab] = useState(0);
+  const { height, ref } = useResizeDetector();
+  const tabsClassName = "tree-widget-tabs";
+  const [heightOfTreeWidget, setHeightOfTreeWidget] = useState(0);
+  const tabElements = document.getElementsByClassName(tabsClassName);
```
Comment (saskliutas) [inline apps/test-app/frontend/src/components/tree-widget/TreeWidget.tsx:44]: Don't use this to access native html element. Use `useRef(null)` instead.

**Review by saskliutas — CHANGES_REQUESTED** [review]

**Review by grigasp — CHANGES_REQUESTED** [review]
Please also update `parseTreeNodeItem` function to assign id like this:
```
id: JSON.stringify([...node.parentKeys, node.key]),
```
This should make sure we don't get any duplicate node ids.

File: apps/test-app/frontend/src/components/tree-widget/TreeWidget.tsx (hunk 1 above)
Comment (JonasDov) [reply apps/test-app/frontend/src/components/tree-widget/TreeWidget.tsx]: Removed useResizeDetector usage from rules driven and stateless trees.

File: apps/test-app/frontend/src/components/tree-widget/TreeWidget.tsx (hunk 2 above)
Comment (JonasDov) [reply apps/test-app/frontend/src/components/tree-widget/TreeWidget.tsx]: `<Tabs>`  and `<Tab>` elements don't have `ref` property, so `useRef` can't be used with them. You can set useRef for each `<div className="filteredTree">`, but then, when trying to resize the right panel, "filteredTree" element height gets updated very slowly.
Changed to:
`ref.current?.getElementsByClassName(tabsClassName);` and moved it to the `useEffect` block.

**Comment by JonasDov** [comment]
> Please also update `parseTreeNodeItem` function to assign id like this:
> 
> ```
> id: JSON.stringify([...node.parentKeys, node.key]),
> ```
> 
> This should make sure we don't get any duplicate node ids.

Changed

**Review by saskliutas — APPROVED** [review]

**Review by grigasp — APPROVED** [review]

**Pull request merged**